databases/week1/homework.sql

```diff
@@ -0,0 +1,37 @@
+-- 1. Find out how many tasks are in the task table
+--  Answer: 35
+SELECT COUNT(*) AS tasks_amount FROM task;
+
+-- 2. Find out how many tasks in the task table do not have a valid due date
+--  Answer: 8
+SELECT COUNT(*) AS tasks_with_valid_due_amount FROM task WHERE due_date IS NULL;
+
+-- 3. Find all the tasks that are marked as done
+SELECT task.title AS done_tasks FROM task WHERE status_id = (SELECT id FROM status WHERE name='done'); -- 12 tasks
+
+-- 4. Find all the tasks that are not marked as done
+SELECT task.title AS not_done_tasks FROM task WHERE status_id != (SELECT id FROM status WHERE name='done'); -- 23 tasks
+
+-- 5. Get all the tasks, sorted with the most recently created first
+SELECT * FROM task ORDER BY created DESC;
+
+-- 6. Get the single most recently created task
+SELECT * FROM task ORDER BY created DESC LIMIT 1;
+
+-- 7. Get the title and due date of all tasks where the title or description contains database
+SELECT task.title, task.due_date FROM task WHERE title LIKE '%database%' OR description LIKE '%database%';
+
+-- 8. Get the title and status (as text) of all tasks
+SELECT task.title AS task_title, status.name AS task_status 
+FROM task LEFT JOIN status ON task.status_id = status.id;
```

Comment:
You don't need a left join for this query, a join would be enough (join default is inner join). In this exercise you cant see the difference between them. You can see the main differences here https://www.w3schools.com/mysql/mysql_join.asp